ucdlib-special/views/collection.twig

```diff
@@ -20,25 +20,44 @@
       <div>
     {% endif %}
     
-      <h3>Collection Number: {{post.meta('callNumber')}}</h3>
+    <h3>Collection Number: {{post.meta('callNumber')}}</h3>
+    
+    {% if post.meta('description')|length > 0 %}
       <p>{{post.meta('description')}}</p>
-
+    {% endif %}
+    
+    {% if post.meta('findingAid')|length > 0 %}
       <h4>Finding Aid</h4>
       <a href="{{post.meta('findingAid').linkURL}}" target="_blank">{{post.meta('findingAid').linkTitle}}</a>
-
-      <h4>Biography</h4>
-      <p>{{post.meta('biography')}}</p>
-
+    {% endif %}
+    
+    {% if post.meta('inclusiveDates')|length > 0 %}
       <h4>Inclusive Dates</h4>
       <p>{{post.meta('inclusiveDates')}}</p>
+    {% endif %}
 
+    {% if post.meta('extent')|length > 0 %}
       <h4>Extent</h4>
       <p>{{post.meta('extent')}}</p>
+    {% endif %}
 
+
+    {% if post.terms('collection-subject')|length > 0 %}
       <h4>Subject</h4>
-      {% for sub in post.meta('subject') %}
-        <p>{{sub}}</p>
+      {{ post.terms('collection-subject')|join('<br>') }}
+    {% endif %}
+
+      {% set headerRendered = false %}
+      {% for col in post.meta('links') %}
+        {% if col.linkType == 'referenceInfo' and col.linkURL|length > 0 and col.displayLabel|length > 0 %}
```

Comment:
To simplify this display logic/make your twigs more readable, move this to your [post class](https://github.com/UCDavisLibrary/ucdlib-wp-plugins/blob/stage/ucdlib-special/includes/collection.php#L168). 

so in your new `referenceInfoLinks` method, you would filter your `post.meta('links')` and then in your twig you could just do:
```twig
{% if post.referenceInfoLinks %}
  <h4>Additional Collection Info</h4>
  {% for link in post.referenceInfoLinks %}
    <p><a href='{{link.linkURL}}'>{{link.displayLabel}}</a></p>
  {% endfor %}
{% endif %}
```
instead of this `headerRendered` thing.

Also, it looks like you already set up meta getter methods in your post class, so stuff like `{{post.meta('extent')}}` should just be `{{post.extent}}` . That way, you know it is [getting cached](https://github.com/UCDavisLibrary/ucdlib-wp-plugins/blob/stage/ucdlib-special/includes/collection.php#L393) and you are only doing one db transaction.

Reply:
Thanks @spelkey-ucd! These changes have been committed.